### PR TITLE
adding server address to qrcode

### PR DESCRIPTION
**Pull request by eoliveros**
added the server to the qrcode. seems to be working. i could see the address after it was parse in the qrcode.

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -728,7 +729,9 @@ class ApiKeyModelView(BaseOnlyUserOwnedModelView):
     def _format_qrcode(view, context, model, name):
         admin = model.account_admin if model.account_admin else False
         address = model.user.wallet_address if model.user.wallet_address else ''
-        data = 'zapm_apikey:%s?secret=%s&name=%s&admin=%r&address=%s' % (model.token, model.secret, model.name, admin, address)
+        url = urlparse(request.base_url)
+        server = url.scheme+'://'+url.netloc.split(":")[0]
```
Comment (djpnewton) [inline models.py]: dont strip out the port part of the netloc (if there is a port specified in the url then we should use it, also the code as is will crash if there is no port specified)

also proper python string formatting (https://pyformat.info/#simple)

**Review by djpnewton — CHANGES_REQUESTED** [review]
Put a '/' at the end of the server param

Then can you or push this to the test server and I will test with the app

**Comment by eoliveros** [comment]
just pushed the change.

File: models.py [hunk 2]
```diff
@@ -728,7 +729,9 @@ class ApiKeyModelView(BaseOnlyUserOwnedModelView):
     def _format_qrcode(view, context, model, name):
         admin = model.account_admin if model.account_admin else False
         address = model.user.wallet_address if model.user.wallet_address else ''
-        data = 'zapm_apikey:%s?secret=%s&name=%s&admin=%r&address=%s' % (model.token, model.secret, model.name, admin, address)
+        url = urlparse(request.base_url)
+        server = '{}{}{}{}'.format(url.scheme, '://', url.netloc, '/')
```
Comment (djpnewton) [inline models.py]: this is much better expressed as:
`server = '{}://{}/'.format(url.scheme, url.netloc)`

**Review by djpnewton — CHANGES_REQUESTED** [review]
also let me know when you deploy to the test server

**Comment by eoliveros** [comment]
somethings not right.
failing when build the image at this line:
```
Building wheel for gevent (setup.py): started
```

and starts erroring:
```
Something bad happened while retrieving test-zapm-server app build logs.
Error: socket hang up
⠧ Building your source code...

Something bad happened while retrieving test-zapm-server app build logs.
Error: connect ECONNREFUSED 157.245.188.90:443
⠸ Building your source code...
```

Not sure whats wrong

**Comment by eoliveros** [comment]
deleting images. and trying again

**Comment by eoliveros** [comment]
i think something has changed thats affecting the building of the container. Seems to affect gevent. When it gets to building this, it fails in this process, will check and see if there are others having the same problem.

**Comment by eoliveros** [comment]
nope can't get this to build again. definitely think there was an update somewhere and its now affecting the gevent module from.

--- looks like its buuild but now its missing:
ModuleNotFoundError: No module named 'ellipticcurve'


Somtehing definitely updated cauign this issues.

**Comment by djpnewton** [comment]
try this: https://github.com/sendgrid/sendgrid-python/issues/913#issuecomment-649074580

File: models.py (hunk 2 above)
Comment (eoliveros) [reply models.py]: changed it.

**Comment by eoliveros** [comment]
added and  managed to deploy it and access the site again.

**Comment by djpnewton** [comment]
tested, for some reason the qr code generated has http not https for the server parameter (even though the site I am accessing is https://....)

You might want to change from request.base_url to request.path (request.path i think comes directly from the client http request so should have the same scheme that the client is using, base_url looks like it is built from various server settings and the flask server is actually sitting behind and proxy server like apache so wont be providing https directly)

**Comment by eoliveros** [comment]
request.path only returns whats after the base_url. trying request.url.

**Comment by eoliveros** [comment]
still http instead of https. i think it maybe because internally it is http. hence when begin queried its returning http as the scheme?

if that's the case, I could rewrite teh server to in the "s" at the end of the scheme?

**Comment by djpnewton** [comment]
hmm.. you are right, we need to detect if the app is behind a proxy

you need to do something like:

```
scheme = url.scheme
if 'X-Forwarded-Proto' in request.headers:
  scheme = request.headers['X-Forwarded-Proto']
```

and use that scheme value, that means that if there is a reverse proxy server and it sets that header (which I believe caprover does by default) the app will use that original scheme instead

**Comment by eoliveros** [comment]
pushed the new commit.

**Pull request merged**